src/ALCProxy.Communication/IProxyServer.cs

```diff
@@ -0,0 +1,157 @@
+            return currentLoadContext.LoadFromAssemblyPath(assemblyPath).GetType(toConvert.FullName);
+        }
+        public object CallObject(MethodInfo targetMethod, IList<object> streams, IList<Type> argTypes)
+        {
+            //Turn the memstreams into their respective objects
+            argTypes = argTypes.Select(x => ConvertType(x)).ToList();
+            object[] args = DeserializeParameters(streams, argTypes);
+            MethodInfo[] methods = instance.GetType().GetMethods();
+            MethodInfo m = FindMethod(methods, targetMethod, argTypes.ToArray());
+            if (m.ContainsGenericParameters)
+            {
+                //While this may work without the conversion, we want it to uphold the type-load boundary, don't let the passed in method use anything from outside the target ALC
+                m = m.MakeGenericMethod(targetMethod.GetGenericArguments().Select(x => ConvertType(x)).ToArray());
+            }
+            return SerializeReturnObject(m.Invoke(instance, args), m.ReturnType);
+        }
+        /// <summary>
+        /// Searches for methods within the type to find the one that matches our passed in type. Since the types are technically different,
+        /// using a .Equals() on the methods doesn't have the comparison work correctly, so the first if statement does that manually for us.
+        /// </summary>
+        protected MethodInfo FindMethod(MethodInfo[] methods, MethodInfo targetMethod, Type[] parameterTypes/*These have already been converted so no issues with compatibility*/)
+        {
+            string methodName = targetMethod.Name;
+            foreach (MethodInfo m in methods)
+            {
+                if (!m.Name.Equals(methodName) || parameterTypes.Length != m.GetParameters().Length)
+                {
+                    continue;
+                }
+                bool methodParamsAlligned = true;
+                for (int i = 0; i < parameterTypes.Length; i++)
+                {
+                    if (!RecursivelyCheckForTypes(parameterTypes[i], m.GetParameters()[i].ParameterType))
+                    {
+                        methodParamsAlligned = false;
+                        break;
+                    }
+                }
+                if (!methodParamsAlligned)
+                    continue;
+                return m;
+            }
+            throw new MissingMethodException("Error in ALCProxy: Method Not found for " + instance.ToString() + ": " + methodName);
+        }
+        /// <summary>
+        /// If a parameter of a function isn't the direct type that we've passed in, this function should find that the type we've passed is correct.
+        /// </summary>
+        private bool RecursivelyCheckForTypes(Type sentParameterType, Type toCompare)
+        {
+            Type[] interfaces = sentParameterType.GetInterfaces();
+            if (sentParameterType.Equals(toCompare))
```

Comment:
This seems to be using `Type.Equals` which will likely return false if the assembly is loaded twice - or they are slightly different assemblies. Basically we would need a fully string based comparisons throughout...

Reply:
Since both parameters to this method should've gone through the `ConvertType` method, `Type.Equals` should be working as long as the ALC can't load two of the same type (like you said previously above). That said if we change how that load works, then yeah, we'll definitely need to change this.

Reply:
I see - well ConvertType should not rely on where the assemblies are loaded - it just needs to load them through the right ALC. Then it will keep on working regardless of the actual assembly load layout.